plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -694,6 +696,63 @@ public void testTimestampColumnAndTimestampWithTimeZoneConstant()
         }
     }
 
+    @Test
+    public void testAccessDeniedToTable()
+            throws Exception
+    {
+        String user = "test_access_denied_to_table_" + randomTableSuffix();
+        Session userSession = Session.builder(getSession())
+                .setIdentity(Identity.from(getSession().getIdentity())
+                        .withExtraCredentials(ImmutableMap.of(
+                                "postgresql-user", user,
+                                "postgresql-password", user))
+                        .build())
+                .build();
+        try (AutoCloseable ignored = withUser(user)) {
+            assertQuery(userSession, "SHOW TABLES LIKE '%nation%'", "VALUES 'nation'");
+            String accessDeniedErrorMessageRegexp = "(?s)ERROR: permission denied for schema tpch.*";
+            assertQueryFails(userSession, "SELECT * FROM nation", accessDeniedErrorMessageRegexp);
+            // TODO https://github.com/trinodb/trino/issues/6947 - User should not be able to see columns
+            assertQuery(
+                    userSession,
+                    "DESC nation",
+                    "VALUES " +
+                            "('nationkey', 'bigint', '', '')," +
+                            "('name', 'varchar(25)', '', '')," +
+                            "('regionkey', 'bigint', '', '')," +
+                            "('comment', 'varchar(152)', '', '')");
+            // TODO https://github.com/trinodb/trino/issues/6947 - User should not be able to see table definition
+            assertQuery(
+                    userSession,
+                    "SHOW CREATE TABLE nation",
+                    "VALUES " +
+                            "'CREATE TABLE postgresql.tpch.nation (\n" +
+                            "   nationkey bigint,\n" +
+                            "   name varchar(25),\n" +
```

Comment:
Is it because of caching?

Reply:
hm... Caching is disabled will check that.

Reply:
Yes, caching is disabled. Also it should not matter (unless there is a bug), because identity is part of caching key.